[Consumer Group] Race  condition while joining simultaneously to the same group id

The following exception gets triggered when 2 streams try to join with the same group id.

```
org.agrona.concurrent.AgentTerminationException: java.lang.IllegalArgumentException: offset=147021 length=1610612732 not valid for capacity=4195072
        at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:707)
        at org.agrona.core/org.agrona.concurrent.AgentRunner.doDutyCycle(AgentRunner.java:291)
        at org.agrona.core/org.agrona.concurrent.AgentRunner.run(AgentRunner.java:164)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalArgumentException: offset=147021 length=1610612732 not valid for capacity=4195072
        at org.agrona.core/org.agrona.concurrent.UnsafeBuffer.boundsCheckWrap(UnsafeBuffer.java:1674)
        at org.agrona.core/org.agrona.concurrent.UnsafeBuffer.wrap(UnsafeBuffer.java:246)
        at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.types.Array32FW.wrap(Array32FW.java:117)
        at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaCacheServerConsumerFactory$KafkaCacheServerConsumerFanout.onConsumerReplyData(KafkaCacheServerConsumerFactory.java:909)
        at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaCacheServerConsumerFactory$KafkaCacheServerConsumerFanout.onConsumerMessage(KafkaCacheServerConsumerFactory.java:799)
        at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleReadReply(DispatchAgent.java:1244)
        at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleRead(DispatchAgent.java:1045)
        at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.concurent.ManyToOneRingBuffer.read(ManyToOneRingBuffer.java:181)
        at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:701)
        ... 3 more
        Suppressed: java.lang.Exception: [engine/data#9]        [0x0909000000000104] streams=[consumeAt=0x00023e48 (0x0000000000023e48), produceAt=0x00023f08 (0x0000000000023f08)]
                at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:705)
                ... 3 more
stopped
```

--- a/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientGroupFactory.java
+++ b/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientGroupFactory.java
@@ -3378,7 +3378,6 @@ public final class KafkaClientGroupFactory extends KafkaClientSaslHandshaker imp
                     final int offset = 0;
                     final int sizeof = payload.sizeof();
 
-                    encoders.add(encodeSyncGroupRequest);
                     signaler.signalNow(originId, routedId, initialId, traceId, SIGNAL_SYNC_GROUP_REQUEST, 0,
                         buffer, offset, sizeof);
                 }

--- a/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientGroupFactory.java
+++ b/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientGroupFactory.java
@@ -3371,6 +3371,17 @@ public final class KafkaClientGroupFactory extends KafkaClientSaslHandshaker imp
             case SIGNAL_SYNC_GROUP_REQUEST:
                 assignment = payload;
                 doEncodeRequestIfNecessary(traceId, initialBudgetId);
+
+                if (decoder != decodeSyncGroupResponse)
+                {
+                    final DirectBuffer buffer = payload.value();
+                    final int offset = 0;
+                    final int sizeof = payload.sizeof();
+
+                    encoders.add(encodeSyncGroupRequest);
+                    signaler.signalNow(originId, routedId, initialId, traceId, SIGNAL_SYNC_GROUP_REQUEST, 0,
+                        buffer, offset, sizeof);
+                }
                 break;
             case SIGNAL_HEARTBEAT_REQUEST:
                 encoders.add(encodeHeartbeatRequest);

